DigitalLearningSolutions.Web/Controllers/TrackingSystem/Centre/Reports/ReportsController.cs

```diff
@@ -76,5 +77,41 @@ public IEnumerable<ActivityDataRowModel> GetGraphData()
         // {
         //     return View();
         // }
+
+        [HttpGet]
+        [Route("Download")]
+        public IActionResult DownloadUsageData(
+            int? jobGroupId,
+            int? courseCategoryId,
+            int? customisationId,
+            string startDate,
+            string endDate,
+            ReportInterval reportInterval
+        )
+        {
+            var parsedStartDate = DateTime.Parse(startDate);
+            var parsedEndDate = DateTime.Parse(endDate);
+
+            var centreId = User.GetCentreId();
+            var adminId = User.GetAdminId()!.Value;
+            var adminUser = userDataService.GetAdminUserById(adminId)!;
+
+            var filterData = new ActivityFilterData(
+                parsedStartDate,
+                parsedEndDate,
+                jobGroupId,
+                adminUser.CategoryId == 0 ? courseCategoryId : adminUser.CategoryId,
+                customisationId,
+                reportInterval
+            );
+
+            var dataFile = activityService.GetActivityDataFileForCentre(centreId, filterData);
+
+            return File(
+                dataFile,
+                "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet",

```

Comment:
Is this worth storing as a variable somewhere in case other methods need to return spreadsheets?

Reply:
I copied it from somewhere else, so definitely yes. Should have pulled it out as a constant at the time.